src/Package/Psr4Info.php

```diff
@@ -0,0 +1,205 @@
+     * @return string
+     */
+    public function getClassNamespaceFromPath(string $path): string
+    {
+        $namespace = ($this->filterDirectoryToNamespace)($this->normalizePath($path));
+
+        return $this->normalizeNamespace($this->getPackagePrefix() . '\\' . $namespace);
+    }
+
+    public function getFullyQualifiedClassNameFromFilename(string $filename): string
+    {
+        [$path, $name] = $this->getPathAndNameFromFilename($filename);
+
+        return $this->getClassNamespaceFromPath($path) . '\\' . $name;
+    }
+
+    public function getClassNamespace(string $fcqn): string
+    {
+        $namespace = $this->normalizeNamespace($fcqn);
+        $namespace = \substr($namespace, 0, \strrpos($namespace, '/'));
+
+        return $this->normalizeNamespace($this->getPackagePrefix() . '\\' . $namespace);
+    }
+
+    public function getClassName(string $fqcn): string
+    {
+        $fqcn = $this->normalizeNamespace($fqcn);
+
+        return \trim(\substr($fqcn, \strrpos($fqcn, '\\')), '\\');
+    }
+
+    public function getPath(string $fqcn): string
+    {
+        $fqcn = $this->normalizeNamespace($fqcn);
+        $namespace = \str_replace($this->getPackagePrefix(), '', $fqcn);
+        $namespace = \ltrim(\substr($namespace, 0, \strrpos($namespace, '\\')), '\\');
+
+        return ($this->filterNamespaceToDirectory)($namespace);
+    }
+
+    public function getFilenameFromPathAndName(string $path, string $name): string
+    {
+        $filePath = $this->getSourceFolder() . DIRECTORY_SEPARATOR;
+
+        if ($path = \trim($path, '/')) {
+            $filePath .= $this->normalizePath($path) . DIRECTORY_SEPARATOR;
+        }
+
+        return $filePath . $name . '.php';
+    }
+
+    public function getPathAndNameFromFilename(string $filename): array
+    {
+        $path = \substr($filename, 0, \strrpos($filename, DIRECTORY_SEPARATOR));
+        $name = \substr($filename, \strrpos($filename, DIRECTORY_SEPARATOR) + 1);
+        $name = \substr($name, 0, \strpos($name, '.') ?: \strlen($name));
+
+        return [$this->normalizePath($path), $name];
+    }
+
+    public function isValidPath(string $filenameOrPath): bool
+    {
+        $path = \substr($filenameOrPath, 0, \strrpos($filenameOrPath, DIRECTORY_SEPARATOR));
+
+        if (0 === \strpos($path, $this->sourceFolder)) {
+            return true;
+        }
+
+        return false;
+    }
+
+    public function getSourceFolder(): string
+    {
+        return $this->sourceFolder;
+    }
+
+    /**
+     * Removes duplicates of backslashes and trims backslashes
+     *
+     * @param string $namespace
+     * @return string
+     */
+    private function normalizeNamespace(string $namespace): string
+    {
+        $namespace = \str_replace('\\\\', '\\', $namespace);
+        $splitNamespace = \explode('\\', $namespace);
+
+        return \trim(\implode('\\', $splitNamespace), '\\');
+    }
+
+    /**
+     * PSR-4 folders must be upper camel case
```

Comment:
Is this true according to the PSR-4 standard?

Reply:
No, it's only case sensitive.